src/Node/TypesNode.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Twig\Node;
+
+use Twig\Attribute\YieldReady;
+use Twig\Compiler;
+use Twig\Node\Expression\ArrayExpression;
+use Twig\Node\Expression\ConstantExpression;
+use Twig\Node\Expression\NameExpression;
+
+/**
+ * Represents a types node.
+ *
+ * @author Jeroen Versteeg <jeroen@alisqi.com>
+ * @see    https://github.com/twigphp/Twig/issues/4165
+ */
+#[YieldReady]
+class TypesNode extends Node implements NodeCaptureInterface
+{
+    public function __construct(ArrayExpression $typesNode, int $lineno)
+    {
+        $this->validateMapping($typesNode);
```

Comment:
I don't think this needs to be validated anymore, as parseNameToStringMapping already handles this.